### PR TITLE
Deprecate `nGram` and `edgeNGram` names for ngram filters

### PR DESCRIPTION
**Pull request by cbuescher**
The camel case name `nGram` should be removed in favour of `ngram` and
similar for `edgeNGram` and `edge_ngram`. Before removal, we need to
deprecate the camel case names first. This change adds deprecation
warnings for indices with versions 6.4.0 and higher and logs deprecation
warnings.

### CODE REVIEW

**Comment by cbuescher** [comment]
This is quiet similar to #27429, so maybe @jpountz would be a good person to review this since we already had a longer discussion on that topic.

File: modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/CommonAnalysisPlugin.java [hunk 1]
```diff
@@ -237,9 +237,14 @@
         filters.add(PreConfiguredTokenFilter.singleton("dutch_stem", false, input -> new SnowballFilter(input, new DutchStemmer())));
         filters.add(PreConfiguredTokenFilter.singleton("edge_ngram", false, input ->
                 new EdgeNGramTokenFilter(input, EdgeNGramTokenFilter.DEFAULT_MIN_GRAM_SIZE, EdgeNGramTokenFilter.DEFAULT_MAX_GRAM_SIZE)));
-        // TODO deprecate edgeNGram
-        filters.add(PreConfiguredTokenFilter.singleton("edgeNGram", false, input ->
-                new EdgeNGramTokenFilter(input, EdgeNGramTokenFilter.DEFAULT_MIN_GRAM_SIZE, EdgeNGramTokenFilter.DEFAULT_MAX_GRAM_SIZE)));
+        filters.add(PreConfiguredTokenFilter.singletonWithVersion("edgeNGram", false, (reader, version) -> {
+            if (version.onOrAfter(org.elasticsearch.Version.V_6_4_0)) {
+                DEPRECATION_LOGGER.deprecatedAndMaybeLog("esgeNGram_deprecation",
```
Comment (jimczi) [inline modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/CommonAnalysisPlugin.java]: nit: s/esgeNgram/edgeNgram/

**Review by jimczi — APPROVED** [review]
LGTM

**Comment by cbuescher** [comment]
@jimczi thanks, will merge as soon as CI passes again.

**Comment by cbuescher** [comment]
test this please

**Comment by cbuescher** [comment]
@elasticmachine test this please

**Pull request merged**